### ISSUE TITLE
[Bug]: Show a deploy button on Deploy tab [git]

### ISSUE THREAD

**Issue by IAmAnubhavSaini** [appsmithorg/appsmith#13642]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I connect app to git repo, all of the files are getting written to remote repo in one commit. After connection, we take users to deploy tab. Since there are no changes to be committed, we disable commit functionality. And now there is no way for users to Deploy from the tab itself.

Either of one solution would suffice:

1. User should see a deploy button after initial commit. OR
2. We should not take user to Deploy tab after initial connection.

### Steps To Reproduce

1. Create a new app
2. Git connect it to a new remote repo
3. See that user ends up deploy tab, but there is nothing for user to do.

---

## what user sees after initial connection.

![Screenshot 2022-05-06 at 1 41 35 PM](https://user-images.githubusercontent.com/1573771/167094293-f20c39e7-7dc1-476d-8994-06c1f0228df9.png)

### Version

Release

**Comment by IAmAnubhavSaini** [comment]
@Nikhil-Nandagopal @vuiets Let me know your thoughts.

**Comment by Nikhil-Nandagopal** [comment]
@IAmAnubhavSaini what do you think of users being in the committed state after this? Where it says all your changes have b committed and the link to the deployed app is shown below?

**Comment by IAmAnubhavSaini** [comment]
1. We were showing `Commit & Push` button but it was actually `Commit & Push & Deploy`. Now, since we commit and push on connect, deploy stays pending. We don't show Deployed information, because we didn't deploy it. (Let me get more information on this from backend)
2. We can show a toast when the application get connected, that it has been pushed to remote repo too.

Now, one solution is to show a Deploy button instead of showing disabled-commit input and button, when user lands on this tab after connection. I like this one, because for the first time we will see a Deploy button on Deploy tab.

**Comment by Nikhil-Nandagopal** [comment]
Wait commit and push was always meant to signify deploy as well. Because even in our code once we commit, a DP is auto-created just like for any CI/CD. Is this changing? @abhvsn

**Comment by abhvsn** [comment]
@Nikhil-Nandagopal we are deploying whenever we do commit and push. What is changed now is earlier we used to ask user to commit the application files after the connect went through, but now we are doing deploy, commit and push during the connect flow itself

cc: @IAmAnubhavSaini

**Comment by abhvsn** [comment]
@IAmAnubhavSaini on the deploy tab we are making a get status call, which now responds with no changes to commit but we should have the current deployed CTA below the commit and push CTA. 
Just for more context if useful we are sending the application object as a response for connect API where we have `lastDeployedAt` field which can be used to display the time.


<img width="558" alt="Screenshot 2022-05-06 at 2 24 52 PM" src="https://user-images.githubusercontent.com/41686026/167100095-d623b5e6-9798-48ba-977e-a8b48f376bd0.png">

<img width="558" alt="Screenshot 2022-05-06 at 2 25 16 PM" src="https://user-images.githubusercontent.com/41686026/167100163-bf563b85-cb41-4bb2-9fde-70d691e34249.png">

**Comment by IAmAnubhavSaini** [comment]
@Nikhil-Nandagopal I wasn't aware of that. 

---

This is weird behaviour then, just now I disconnected and reconnected and saw the deployed info. But it said 42 minutes ago. Which means, it was deployed when I took original screenshot.

![Screenshot 2022-05-06 at 2 23 09 PM](https://user-images.githubusercontent.com/1573771/167100561-2aab21df-57fa-473f-87e2-715f49f46e0c.png)


Then after a hard refresh, I saw 2 minutes, which was previous connection.

![Screenshot 2022-05-06 at 2 25 31 PM](https://user-images.githubusercontent.com/1573771/167100597-63c678ad-6e61-4831-a2ea-fcbcaab48acc.png)


Will investigate it further.
